expirationd.lua

```diff
@@ -327,48 +390,202 @@ end
+-- @number[opt] options.full_scan_delay
+--     Sleep time between full scans (in seconds). It is allowed to pass an FFI
+--     number: `1LL`, `1UL` etc. Default value is 1 sec.
+-- @number[opt] options.full_scan_time
+--     Time required for a full index scan (in seconds). It is allowed to pass
+--     an FFI number: `1LL`, `1UL` etc. `full_scan_time` used for calculation
+--     of time during which fiber sleeps between iterations. Default value is
+--     3600.
+-- @string[opt] options.index
+--     Name or id of the index to iterate on. If omitted, will use the primary
+--     index. If there's no index with this name, will throw an error.
+--     Supported index types are TREE and HASH, using other types will result
+--     in an error.
+-- @func[opt] options.iterate_with
+--     Function which returns an iterator object which provides tuples to
+--     check, considering the `start_key`, `process_while` and other options.
+--     There's a default function which can be overriden with this parameter.
+--
+-- Example of function:
+--
+-- ```
+-- local function iterate_with()
+--     return true
+-- end
+-- ```
+-- @number[opt] options.iteration_delay
+--     Max sleep time between iterations (in seconds). It is allowed to pass
+--     an FFI number: `1LL`, `1UL` etc. Default value is 1 sec.
+--     Fiber sleeps min(`tuples_per_iteration` × `full_scan_time` / `space_length`, `iteration_delay`).
+-- @string[opt] options.iterator_type
+--     Type of the iterator to use, as string or box.index constant, for
+--     example, `EQ` or `box.index.EQ`, default is `box.index.ALL`. See more
+--     about index iterators in [index_object:pairs()][1].
+--
+--     [1]: https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_index/pairs/.
+--
+-- @func[opt] options.on_full_scan_complete
+--     Function to call after completing a full scan iteration. Default value
+--     is a function that do nothing.
+-- @func[opt] options.on_full_scan_error
+--     Function to call after terminating a full scan due to an error. Default
+--     value is a function that do nothing.
+--
+-- Example of function:
+--
+-- ```
+-- local function on_full_scan_complete()
+--     pcall(fiber.sleep, 1)
+-- end
+-- ```
+-- @func[opt] options.on_full_scan_start
+--     Function to call before starting a full scan iteration. Default value
+--     is a function that do nothing.
+-- @func[opt] options.on_full_scan_success
+--     Function to call after successfully completing a full scan iteration.
+--     Default value is a function that do nothing.
+-- @func[opt] options.process_expired_tuple
+--     Applied to expired tuples, receives `space_id`, `args`, `tuple` as
+--     arguments. When `process_expired_tuple` is not passed (or `nil` passed),
+--     tuples are removed.
+--
+-- Example of function:
+--
+-- ```
+-- local function put_tuple_to_archive(space_id, args, tuple)
+--     box.space[space_id]:delete{tuple[1]}
+--     local email = get_field(tuple, 2)
```

Comment:
`tuple[1]` is used just above, but then `get_field(tuple, 2)`. Why? What is the difference?

Reply:
Fixed.
```lua
@@ -513,7 +517,7 @@ end
 -- ```
 -- local function put_tuple_to_archive(space_id, args, tuple)
 --     box.space[space_id]:delete{tuple[1]}
---     local email = get_field(tuple, 2)
+--     local email = tuple[2]
 --     if args.archive_space_id ~= nil and email ~= nil then
 --         box.space[args.archive_space_id]:replace{email, fiber.time()}
 --     end
```


It was copy-pasted from TAP tests and `get_field` handle cases when tuple is `nil` field_no is bigger than `#tuple`:
```lua
-- get field                                                                     
local function get_field(tuple, field_no)                                        
    if tuple == nil then                                                         
        return nil                                                               
    end                                                                          
                                                                                 
    if #tuple < field_no then                                                    
        return nil                                                               
    end                                                                          
                                                                                 
    return tuple[field_no]                                                       
end                 
```

We don't need get_field< so I removed it.